pkg/controllers/updaterun/initialization_integration_test.go

```diff
@@ -1056,28 +1082,62 @@ func generateSucceededInitializationStatusForSmallClusters(
 		},
 	}
 	for i := range status.StagesStatus {
-		var tasks []placementv1beta1.StageTaskStatus
+		var beforeTasks []placementv1beta1.StageTaskStatus
+		for _, task := range updateStrategy.Spec.Stages[i].BeforeStageTasks {
+			taskStatus := placementv1beta1.StageTaskStatus{Type: task.Type}
+			if task.Type == placementv1beta1.StageTaskTypeApproval {
+				taskStatus.ApprovalRequestName = fmt.Sprintf(placementv1beta1.BeforeStageApprovalTaskNameFmt, updateRun.Name, status.StagesStatus[i].StageName)
+			}
+			beforeTasks = append(beforeTasks, taskStatus)
+		}
+		status.StagesStatus[i].BeforeStageTaskStatus = beforeTasks
+
+		var afterTasks []placementv1beta1.StageTaskStatus
 		for _, task := range updateStrategy.Spec.Stages[i].AfterStageTasks {
 			taskStatus := placementv1beta1.StageTaskStatus{Type: task.Type}
 			if task.Type == placementv1beta1.StageTaskTypeApproval {
-				taskStatus.ApprovalRequestName = updateRun.Name + "-" + status.StagesStatus[i].StageName
+				taskStatus.ApprovalRequestName = fmt.Sprintf(placementv1beta1.AfterStageApprovalTaskNameFmt, updateRun.Name, status.StagesStatus[i].StageName)
 			}
-			tasks = append(tasks, taskStatus)
+			afterTasks = append(afterTasks, taskStatus)
 		}
-		status.StagesStatus[i].AfterStageTaskStatus = tasks
+		status.StagesStatus[i].AfterStageTaskStatus = afterTasks
 	}
 	return status
 }
 
 func generateExecutionStartedStatus(
 	updateRun *placementv1beta1.ClusterStagedUpdateRun,
-	initialized *placementv1beta1.UpdateRunStatus,
+	status *placementv1beta1.UpdateRunStatus,
 ) *placementv1beta1.UpdateRunStatus {
 	// Mark updateRun execution has started.
-	initialized.Conditions = append(initialized.Conditions, generateTrueCondition(updateRun, placementv1beta1.StagedUpdateRunConditionProgressing))
+	if meta.FindStatusCondition(status.Conditions, string(placementv1beta1.StagedUpdateRunConditionProgressing)) != nil {
```

Comment:
We don't need if else here, SetStatusCondition can handle existing, new conditions